File: src/main/java/com/jankalyan/admin/service/AdminService.java
package com.jankalyan.admin.service;

import com.jankalyan.admin.dto.request.UpdateComplaintStatusRequest;
import com.jankalyan.admin.dto.response.AdminDashboardResponse;
import com.jankalyan.complaint.dto.response.ComplaintResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface AdminService {
    
    AdminDashboardResponse getDashboard();
    
    Page<ComplaintResponse> getComplaints(String search, UUID categoryId, com.jankalyan.complaint.entity.ComplaintStatus status, Pageable pageable);

    com.jankalyan.admin.dto.response.AdminComplaintDetailResponse getComplaintDetailsForAdmin(UUID complaintId);
    
    void updateComplaintStatus(UUID complaintId, UpdateComplaintStatusRequest request);
    
    void deleteComplaint(UUID complaintId);
}

File: frontend/src/services/adminService.js
import api from './api';

export const adminService = {
  getDashboard: async () => {
    const response = await api.get('/admin/dashboard');
    return response.data;
  },
  getComplaints: async (page = 0, size = 10) => {
    const response = await api.get('/admin/complaints', { params: { page, size } });
    return response.data;
  },
  updateComplaintStatus: async ({ complaintId, status, comments }) => {
    const response = await api.patch(`/admin/complaints/${complaintId}/status`, { status, comments });
    return response.data;
  },
  deleteComplaint: async (complaintId) => {
    const response = await api.delete(`/admin/complaints/${complaintId}`);
    return response.data;
  }
};
